src/sources/journald.rs

```diff
@@ -331,21 +344,18 @@ where
     J: JournalSource,
     T: Sink<SinkItem = Record, SinkError = ()>,
 {
-    pub fn run(mut self) {
-        let timeout = Duration::from_millis(500); // arbitrary timeout
+    pub async fn run(mut self) {
         let channel = &mut self.channel;
-        let mut shutdown = self.shutdown.compat();
 
         loop {
             let mut saw_record = false;
-            let mut at_end = false;
             let mut cursor: Option<String> = None;
 
             for _ in 0..self.batch_size {
-                let text = match self.journal.next() {
+                let text = match self.journal.next().await {
                     None => {
-                        at_end = true;
-                        break;
+                        let _ = self.shutdown.compat().await;
```

Comment:
@fanatid As I understand, `shutdown` is a signal to stop all sources, I'm not sure if this source should wait for it instead of terminating immediately?

Reply:
It's a good question. I only removed `spawn_blocking` and did not change logic. I think we should return on `None` and I also do not see that we actually use the `shutdown` signal.

Reply:
This wait on `shutdown` seem to be used as backoff originally.

https://github.com/timberio/vector/blob/0f72a2b1669a97e4838d3ca852d2f68a878915f4/src/sources/journald.rs#L171-L175

We use `shutdown` in `select(...)` so should be safe to remove it.

https://github.com/timberio/vector/blob/521711ee44b1dd83426fd8f4ae9f0b90e75769ab/src/sources/journald.rs#L200-L202